docs: New Console features for 23.1 (#1206)
redpanda-data/documentation: docs/manage/security/console/authorization.mdx

The editor role grants all permissions that come with the `viewer` role and additionally includes:

> - Managing all topic aspects (create topic, edit topic configs, delete topic, publish and delete topic records, ...)
> - Managing all cluster configuration aspects (editing node or cluster configs, ...)
> - Managing all consumer group aspects (edit group offsets, delete group offsets)
> - Managing all Kafka connect aspects (create/update/delete or start/pause/stop connectors)

Author: ```suggestion
- Managing all topic aspects, such as creating topics, editing topic configuration, deleting topics, and publishing and deleting topic records.
- Managing all cluster configuration aspects, such as editing node or cluster configs.
- Managing all consumer group aspects, such as editing group offsets or deleting group offsets.
- Managing all Kafka connect aspects, such as creating/updating/deleting or starting/pausing/stopping connectors.
```
